src/test/spec/v2_runner/operation.rs

```diff
@@ -1493,6 +1572,59 @@ impl TestOperation for AssertIndexNotExists {
     }
 }
 
+#[derive(Debug, Deserialize)]
+pub(super) struct WithTransaction {
+    callback: WithTransactionCallback,
+    options: Option<TransactionOptions>,
+}
+
+#[derive(Debug, Deserialize)]
+struct WithTransactionCallback {
+    operations: Vec<Operation>,
+}
+
+impl TestOperation for WithTransaction {
+    fn execute_recursive<'a>(
+        &'a self,
+        runner: &'a mut OpRunner,
+        sessions: OpSessions<'a>,
+    ) -> BoxFuture<'a, Result<Option<Bson>>> {
+        async move {
+            let session = sessions.session0.unwrap();
+            session
+                .with_transaction(
+                    (runner, &self.callback.operations, sessions.session1),
+                    |session, (runner, operations, session1)| {
+                        async move {
+                            for op in operations.iter() {
+                                let sessions = OpSessions {
+                                    session0: Some(session),
```

Comment:
This is literally the same pointer as `session0` passed to `execute_recursive`, but to satisfy the borrow checker we need to re-bind it with the lifetime provided to the callback.